### PR TITLE
Add tmxparser

### PR DESCRIPTION
**Pull request by PucklaMotzer09**
Adds [tmxparser](https://github.com/sainteos/tmxparser)

### CODE REVIEW

File: packages/t/tmxparser/xmake.lua [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+        io.gsub("include/Tmx.h.in", "@VERSION_MINOR@", "2")
+        io.gsub("include/Tmx.h.in", "@VERSION_PATCH@", "0")
+        os.mv("include/Tmx.h.in", "include/Tmx.h")
+        io.writefile("xmake.lua", [[
+            add_requires("zlib", "tinyxml2")
+            add_rules("mode.debug", "mode.release")
+            target("tmxparser")
+                set_kind("$(kind)")
+                add_packages("zlib", "tinyxml2")
+                add_headerfiles("include/**.h")
```
Comment (waruqi) [inline packages/t/tmxparser/xmake.lua:22]: I think that we need only export `add_headerfiles("include/*.h")`

File: packages/t/tmxparser/xmake.lua [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+    set_urls("https://github.com/sainteos/tmxparser.git")
+    add_versions("2.2.0", "d314b3115c7ed86a939eefcb6009a495f043a346")
+
+    add_deps("zlib", "tinyxml2")
+
+    on_install(function (package)
+        io.gsub("include/Tmx.h.in", "@VERSION_MAJOR@", "2")
+        io.gsub("include/Tmx.h.in", "@VERSION_MINOR@", "2")
+        io.gsub("include/Tmx.h.in", "@VERSION_PATCH@", "0")
+        os.mv("include/Tmx.h.in", "include/Tmx.h")
```
Comment (waruqi) [inline packages/t/tmxparser/xmake.lua:15]:
```lua
        io.gsub("include/Tmx.h.in", "@VERSION_PATCH@", "@VERSION_ALTER@")
        io.writefile("xmake.lua", ([[
            set_version("%s")
            add_requires("zlib", "tinyxml2")
            add_rules("mode.debug", "mode.release")
            target("tmxparser")
                set_kind("$(kind)")
                add_packages("zlib", "tinyxml2")
                add_headerfiles("include/*.h")
                add_includedirs("include", "include/base64")
                set_configdir("include")
                add_configfiles("include/Tmx.h.in", {pattern = "@(.-)@"})
                add_files("src/**.cpp")
        ]]):format(package:version_str()))
```

We use `add_configfiles("include/Tmx.h.in", {pattern = "@(.-)@"})` to replace `@VERSION_MAJOR@`, `@VERSION_MINOR@`, `@VERSION_ALTER@` and save to include/Tmx.h

**Comment by PucklaMotzer09** [comment]
The Android build of tinyxml2 fails although it worked before. I think it's because the include directores of libc++ are missing. The following arguments are missing in the current build but were there previously:
+ ```-I/home/runner/work/xmake-repo/xmake-repo/android-ndk-r21/sources/cxx-stl/llvm-libc++/include```
+ ```-I/home/runner/work/xmake-repo/xmake-repo/android-ndk-r21/sources/cxx-stl/llvm-libc++/libs/armeabi-v7a/include```
+ ```-I/home/runner/work/xmake-repo/xmake-repo/android-ndk-r21/sources/cxx-stl/llvm-libc++abi/include```

**Comment by waruqi** [comment]
> > The Android build of tinyxml2 fails although it worked before. I think it's because the include directores of libc++ are missing. The following arguments are missing in the current build but were there previously:
> 
> I adjusted the toolchain for dev branch in the afternoon, which also caused some problems. I will look at it.

I have fixed it, you can try it again.

**Pull request merged**